### PR TITLE
Fix signing by moving ItemsToSign evaluation to be inside a target

### PR DESCRIPTION
**Pull request by safern**
MSBuild had a regresion: https://github.com/microsoft/msbuild/issues/5445 which is causing:

```xml
    <ItemGroup>
      <!-- External files -->
      <ItemsToSign Remove="@(ItemsToSign->WithMetadataValue('Filename', 'Newtonsoft.Json'))" />
    </ItemGroup>
```
To remove all items from the itemgroup, hence not signing anything.

Also, just set allow empty sign list to true if SignFinalPackages != true to catch this earlier, not until signing validation which was moved out of the build already.

Test official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=694513&view=results

cc: @rainersigwald @mmitche

### CODE REVIEW

File: eng/Signing.props [hunk 1]
```diff
@@ -5,114 +5,119 @@
...
+      <ItemsToSign Include="$(DownloadDirectory)**\*.rpm" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <!-- External files -->
+      <ItemsToSign Remove="@(ItemsToSign->WithMetadataValue('Filename', 'Newtonsoft.Json'))" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <ItemsToSign Update="@(ItemsToSign)" Authenticode="$(CertificateId)" />
```
Comment (jkoritzinsky) [inline eng/Signing.props:117]: Since this is now within a target, the `Update` attribute isn't needed.

Comment (safern) [reply eng/Signing.props:117]: Can we leave this as is to keep it minimal? I expect to just revert the target once we consume the fix.

**Review by ViktorHofer — APPROVED** [review]

File: eng/Signing.props (hunk 1 above)
Comment (jkoritzinsky) [reply eng/Signing.props:117]: I believe so, but @rainersigwald can confirm.

**Comment by mmitche** [comment]
What other places might this regression hit that we haven't seen yet?

**Review by mmitche — APPROVED** [review]

**Comment by safern** [comment]
> What other places might this regression hit that we haven't seen yet?

Any places where we do a Remove or Update with item transformation. I can do a quick grep in the repo to see where we do a transformation of items to Remove or Update.

**Comment by safern** [comment]
I downloaded the produced packages on the test build and ran SignValidation locally and it all looks good:

```
No signing issues found
Total Time: 00:06:14.2232334
Total Files: 14079, Signed: 9343, Unsigned: 0, Skipped: 4636, Excluded: 100, Skipped & Excluded: 0
```

**Review by rainersigwald — COMMENTED** [review]
Belated LGTM. There may be another instance with `Remove` but the `Update` class of the bug wasn't a regression and is unlikely to be a source of bugs in runtime.

**Pull request merged**